metadata-service/plugin/src/main/java/com/datahub/plugins/auth/pojo/AuthPluginConfig.java

```diff
@@ -0,0 +1,27 @@
+package com.datahub.plugins.auth.pojo;
+
+import com.datahub.plugins.common.PluginConfigImpl;
+import com.datahub.plugins.common.PluginConfigWithJar;
+import com.datahub.plugins.common.PluginType;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper=false)
+public class AuthPluginConfig extends PluginConfigImpl implements PluginConfigWithJar {
+  private Path pluginHomeDirectory;
+  private Path pluginJarPath;
+
+  public AuthPluginConfig(PluginType type, String name, Boolean enabled, String className,
+      Path pluginHomeDirectory, Path pluginJarPath, Optional<Map<String, Object>> configs) {
+    super(type, name, enabled, className, configs);
+    this.pluginHomeDirectory = pluginHomeDirectory;
```

Comment:
Which arguments are nullable vs not nullable? Consider adding @Nonnull and @Nullable annotations, and then validating those arguments which must not be null using Objects.requireNonNull(arg)